src/olympia/addons/models.py

```diff
@@ -1043,33 +1043,35 @@ def get_default_icon_url(self, size):
         )
 
     @write
-    def update_status(self, ignore_version=None):
+    def update_status(self, ignore_version=None, force_update=False):
         self.reload()
 
-        if (self.status in [amo.STATUS_NULL, amo.STATUS_DELETED] or
-                self.is_disabled or self.is_persona()):
+        if (self.status in [amo.STATUS_DISABLED, amo.STATUS_DELETED] or
+                self.disabled_by_user or self.is_persona()):
             self.update_version(ignore=ignore_version)
```

Comment:
Since we're touching the method, might as well comment where needed. I feel this is one of the places that require one :)